project_stock/models/project_task.py

```diff
@@ -118,11 +118,11 @@ def _prepare_procurement_group_vals(self, values):
         return {"name": values["name"]}
 
     def action_confirm(self):
-        self.mapped("move_ids").write({"state": "confirmed"})
-        self.action_assign()
+        self.mapped("move_ids")._action_confirm()
+        self.mapped("move_ids")._action_assign()
```

Comment:
Why is this here?

Reply:
This change is related to https://github.com/OCA/project/pull/914#discussion_r854118954.

I think it's good to explain the process to understand everything.

**Before:**
- "Confirm Materials" button confirmed stock moves with the `action_confirm()` method.
- "Check availability" button called the `_action_assign()` method to reserve the stock.
- Additionally, "Confirm materials" button called the method of the "Check availability" button (to avoid clicking on confirm and always check availability).

**Actual use case:**
- The task is changed to "In Progress" stage.
- Stock moves are added and confirmed with "Confirm materials" button.
- The task is changed to "Completed" stage.
- The task is changed back to "In Progress" stage.
- Some extra stock moves is added.
- "Check availability" button does nothing for the new line because the status of line is draft (not confirmed).

**Changes done:**
- [x] Changed `self.mapped("move_ids").write({"state": "confirmed"})` to `self.mapped("move_ids")._action_confirm()`.
- [x] Changed `self.action_assign()` to `self.mapped("move_ids")._action_assign()`.
- [x] Changed the `action_assign()` event to call `action_confirm()` event and solve the previous use case.

Reply:
My comment was not asking for that. There's no sense to have 2 methods that do the same